apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx

```diff
@@ -156,6 +159,15 @@ const SwapForm = () => {
 
   const { setDrawer } = React.useContext(context);
 
+  useEffect(() => {
+    const refreshInterval = setInterval(() => {
+      swapTransaction?.swap?.refetchRates();
+    }, refreshTime);
+    return () => {
+      clearInterval(refreshInterval);
+    };
+  }, [swapTransaction.swap.from.amount]);
```

Comment:
Should this not also include anything from the swap (to, provider, etc)?

Reply:
Ah I see, only the from can be changed, not the 'To', and the whole quote is eventually updated when provider is changed. The only thing is there are a few seconds when the timer for the previous quote still counts down when the provider has changed. 

https://user-images.githubusercontent.com/87659842/189901083-e0559990-0519-4679-ad10-b72a40c4b16e.mov

Reply:
This is the bug that I was talking this morning. 
In the new version, I am restoring the data when the user change "From" or "To".